artifacts.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+const artifactsRouterPath = "/epr/{packageName}/{packageName:[a-z]+}-{packageVersion}.tar.gz"
```

Comment:
I expected that we have some packages with an `_` but couldn't find one: https://github.com/elastic/integrations/tree/master/dev/packages/beats

Reply:
Right, `php_fpm`

Reply:
Sounds like we need to support it then.

We should make sure that our validation code for package names (if we have one) validates for the same pattern. Perhaps we can even reuse it.

Reply:
> Sounds like we need to support it then.

Done

> We should make sure that our validation code for package names (if we have one) validates for the same pattern. Perhaps we can even reuse it.

I don't see any. Do we have it?

Reply:
Looks like we don't :-( https://github.com/elastic/package-registry/blob/master/util/package.go#L264